### ISSUE TITLE
Authentication failure

### ISSUE THREAD

**Issue by ThomasFrisendal** [neo4j-apps/neo4j-desktop#64]
Hi, 

I was working this morning, but now I am getting nowhere.

I am on:
Neo4j Browser 3.2.19
Neo4j 3.5 Enterprise (developer)

When I start a database from the desktop, I get:
![image](https://user-images.githubusercontent.com/12172003/56899067-d30b1880-6a92-11e9-9a28-1c34c7e4e478.png)

I do not believe that any of the databases that I tried have a password on their own.

Trying every pw I can think of, I have to give up and press Ignore, which then, in the Desktop brings me to having to connect to the server (which is running).

![image](https://user-images.githubusercontent.com/12172003/56899184-241b0c80-6a93-11e9-967a-c641e15c789d.png)

Again, I am unable to find an acceptable password, so I have to give up connecting to the database.

In the server log I see:

Starting Neo4j.
2019-04-29 13:14:51.663+0000 WARN  ha.pull_interval is deprecated.
2019-04-29 13:14:51.751+0000 INFO  ======== Neo4j 3.5.0 ========
2019-04-29 13:14:51.770+0000 INFO  Starting...
2019-04-29 13:15:04.434+0000 INFO  Initiating metrics...
2019-04-29 13:15:17.481+0000 INFO  Sending metrics to CSV file at /Users/Thomas/Library/Application Support/Neo4j Desktop/Application/neo4jDatabases/database-67c0cdba-2c5e-4d41-8a72-b03a619c6e55/installation-3.5.0/metrics
2019-04-29 13:15:20.569+0000 WARN  Server thread metrics not available (missing neo4j.server.threads.jetty.all)
2019-04-29 13:15:20.571+0000 WARN  Server thread metrics not available (missing neo4j.server.threads.jetty.idle)
2019-04-29 13:15:28.202+0000 INFO  Bolt enabled on 127.0.0.1:7687.
2019-04-29 13:15:32.118+0000 INFO  Started.
2019-04-29 13:15:32.680+0000 INFO  Mounted REST API at: /db/manage
2019-04-29 13:15:32.939+0000 INFO  Server thread metrics have been registered successfully
2019-04-29 13:15:34.620+0000 INFO  Remote interface available at http://localhost:7474/
2019-04-29 13:15:36.264+0000 WARN  Unsupported authentication token, missing key `scheme`: { user_agent='neo4j-javascript/1.7.2' }
2019-04-29 13:16:11.480+0000 WARN  The client is unauthorized due to authentication failure.
2019-04-29 13:16:11.559+0000 WARN  The client is unauthorized due to authentication failure.

What really puzzles me, is that when I look in my profile in the desktop here is what I see:

![image](https://user-images.githubusercontent.com/12172003/56899311-70fee300-6a93-11e9-8acb-e42b681bd045.png)

I got the activation key by using my GitHub signing in January.

Note the date. It expires one year from now. A mouse-over tells me that it is at 9:39 AM.

I was working this morning (before 9:30) but now I am not able to be authenticated... One year too early?

Please help - I am stuck right now.

Best 
Thomas

**Comment by oskarhane** [comment]
Hi Thomas and thanks for raising this issue.
Sorry for the confusion, but your dbms/graph authentication and  the registration of the Neo4j Desktop application are totally separate.
Each "Graph" in your projects have their own username (usually "neo4j") and a password you needed to set when you created the graph in the first place.

To reset the password for a dbms/graph you'll need to go into the file system (you can get to the dbms folder by clicking "Manage" on a graph and then the "Open Folder" button) and delete the "data/dbms/auth" file and then start the db, open Neo4j Browser and set a new password there.
Neo4j Desktop will then ask you for your newly set password, as in the screenshot you showed.

Once you've entered it there, you will not be prompted for it again in either Desktop or any graph apps.

**Comment by ThomasFrisendal** [comment]
Hi OskarNot at my computer right now. Why did this start to happening today? No new databases.Thomas29. apr. 2019 15.59 skrev Oskar Hane <notifications@github.com>:Hi Thomas and thanks for raising this issue.
Sorry for the confusion, but your dbms/graph authentication and  the registration of the Neo4j Desktop application are totally separate.
Each "Graph" in your projects have their own username (usually "neo4j") and a password you needed to set when you created the graph in the first place.
To reset the password for a dbms/graph you'll need to go into the file system (you can get to the dbms folder by clicking "Manage" on a graph and then the "Open Folder" button) and delete the "data/dbms/auth" file and then start the db, open Neo4j Browser and set a new password there.
Neo4j Desktop will then ask you for your newly set password, as in the screenshot you showed.
Once you've entered it there, you will not be prompted for it again in either Desktop or any graph apps.

**Comment by oskarhane** [comment]
That's what puzzles me. 
What Neo4j Desktop version are you on?
Did you perform any updates to your operating system since last time you used Neo4j Desktop?
We store the graphs username/password in your operating systems secure keychain, so there might be some permissions issues that appeared there (just guessing at this point).
Could you send me your Desktop log file (get to it by pressing ctrl+k and then type "log" and chose desktop log file format he list)? I can be reached at oskar.hane at neo4j.com

**Comment by ThomasFrisendal** [comment]
I think this mess was caused by my upgrade to MAC OS X Mojave some weeks ago. I have two keyrings (one from this computer and one from my previous computer). Mojave did not take this easy. But after a couple of weeks some fix must have arrived, which cleaned up some things. But it also meant that some passwords were forgotten. I did not know that Neo4j also uses the keyring, and I don't understand why there was a delay of some days. But maybe Apple installed a fix by mid day today.
Case closed.
Anyway.

Case is closed - thanks Oskar!
